### ISSUE TITLE
Support Kafka based consumer offsets storage

### ISSUE THREAD

**Issue by akki** [uber/uReplicator#304]
#### Feature request
Currently, uReplicator supports only Zookeeper based offset storage. But the newer versions of Kafka allow (and encourage) to store offsets in Kafka's built-in topic (`__consumer_offset`).

So, It'd be nice if uReplicator allows the option to choose offset storage (`zookeeper` or `kafka`).

#### Related tickets
This would also be one step closer in achieving https://github.com/uber/uReplicator/issues/228.
I also found a similar request in https://github.com/uber/uReplicator/issues/246 but it was closed.

<br />
So I would like to know if the maintainers would be interested to see such a feature?

**Comment by akki** [comment]
From my understanding, to implement this feature one will need to add a `KafkaCheckpointManager` (like [ZookeeperCheckpointManager](https://github.com/uber/uReplicator/blob/master/uReplicator-Worker-3.0/src/main/java/com/uber/stream/ureplicator/worker/ZookeeperCheckpointManager.java)) and hook it into the [createCheckpointManager](https://github.com/uber/uReplicator/blob/master/uReplicator-Worker-3.0/src/main/java/com/uber/stream/ureplicator/worker/WorkerInstance.java#L148) method.

If that is the right direction, I might be able to send a PR for this from one of our projects built on top of uReplicator which already has this feature.

**Comment by yangy0000** [comment]
yes, Kafka based consumer offsets storage is a good feature to have. it would be great if you want to contribute to this

**Comment by cricket007** [comment]
What is the lifecycle of uReplicator with [KIP-382](https://cwiki.apache.org/confluence/display/KAFKA/KIP-382%3A+MirrorMaker+2.0) released?

**Comment by yangy0000** [comment]
We will keep maintaining, fixing bugs and improving the document, but we don't have any major change for uReplicator right now.
